pkgs/applications/virtualization/sail-riscv/default.nix

```
@@ -0,0 +1,48 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+
+, arch
+
+, ocamlPackages
+, ocaml
+, zlib
+, z3
+}:
+
+
+stdenv.mkDerivation rec {
+  pname = "sail-riscv";
+  version = "0.5";
+
+  nativeBuildInputs = with ocamlPackages; [ ocamlbuild findlib ocaml z3 ];
+  buildInputs = with ocamlPackages; [ sail zlib ];
+
+  prePatch = "rm -r prover_snapshots";
+
+  src = fetchFromGitHub {
+    owner = "riscv";
+    repo = pname;
+    rev = version;
+    hash = "sha256-7PZNNUMaCZEBf0lOCqkquewRgZPooBOjIbGF7JlLnEo=";
+  };
+
+  makeFlags = [ "SAIL=sail" "ARCH=${arch}" "SAIL_DIR=${ocamlPackages.sail}/share/sail" ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    cp c_emulator/riscv_sim_${arch} $out/bin
+    mkdir $out/share/
+    cp -r generated_definitions/{coq,hol4,isabelle} $out/share/
+    runHook postInstall
+  '';
```

Keep an empty line between the boilerplate codes and your own.
```suggestion
  installPhase = ''
    runHook preInstall

    mkdir -p $out/bin
    cp c_emulator/riscv_sim_${arch} $out/bin
    mkdir $out/share/
    cp -r generated_definitions/{coq,hol4,isabelle} $out/share/

    runHook postInstall
  '';
```